### DISCUSSION TITLE
Porting over from Adafruit nrf52 board question

### DISCUSSION THREAD

**Discussion by LokiMetaSmith** [h2zero/n-able-Arduino#9] [Q&A]
I'm doing a custom board nrf52833, and I like a lot of what I see in this project, but I'm porting over from Adafruit which has a very similarly structured project, but different enough where it's biting me again and again. 

My question is I'm using littleFS for SoC flash data storage and EEPROM management, SoftwareTimer for non-critical UI events, and the adafruit_nrfCrypto library to use the built in hash functions. There are also a ton of quality of life and opinionated changes in the Adafruit library. So, is porting over this stuff of interest?

I could probably ditch most of it and work with what's already included, I'll do that anyway with porting over from bluefruit to  nimble.

**Comment by h2zero** [comment]
Thank you for opening this discussion!

The more the merrier, this project is an open canvas! I only put the basics in place, it is wide open for contributions. littleFS was on my list to add, but I wanted to get this out and didn't have time to do that. I'm not very familiar with the adafruit core/libraries, so anything you would like to suggest/PR from there, or anywhere else, would be more than welcome. 

My goal with this is to remove barriers and open access to possibilities that don't require vendor/software lock-in while being user friendly. If it maintains/enhances this goal then it will absolutely be welcomed.

**Reply by LokiMetaSmith** [reply]
So I've been taking a crack at doing this, and well it was a bit more work than anticipated. 
https://github.com/LokiMetaSmith/n-able-Arduino

It turns out that adafruit uses an older version of some libraries and n-able-Arduino uses older versions of others. 
Basically it's a hot mess, with how Nordic uses complier defines to flag different patches and cover for older code, so you have to be familiar with FreeRTOS, nordic nrfx libraries, and the tweaks that adafruit did to make their boards work. 

Anyways, posting this as a status update, no where near compiling, but maybe it's a useful starting point.

**Reply by h2zero** [reply]
Yeah, I saw that adafruit made some customizations and didn't want to copy them unless necessary and useful. I don't know the what they currently use for freertos version or nrfx, I think I had a newer version than they did at the time of implementation.

Thanks for the update, I'll have a look at your situation and maybe I can offer some help.

**Reply by h2zero** [reply]
I had a look at what your have done so far, great effort!

I think the issue is trying to do too much at once. Bringing in the desired libraries one at a time, testing, making changes to them as required, repeat, would make this easier I think. For instance, I just pulled in the HardwarePWM and Servo libraries (servo branch) for a project that, with some minor changes, seems to function fine so far.

The question I have is what would you like to see added first? We could work from there by adding that one then the next etc..
